components/kyma-operator/pkg/kymaoperation/steps/factory.go

```diff
@@ -0,0 +1,119 @@
+package steps
+
+import (
+	errors "github.com/pkg/errors"
+
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/apis/installer/v1alpha1"
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/config"
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/kymahelm"
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/overrides"
+)
+
+const (
+	defaultDeleteWaitTimeSec   = 10
```

Comment:
I think those values should be configurable with envs/params

Reply:
This PR focused on more high-level refactoring, I didn't want to add anything new in terms of features/flags/params/etc. Besides that, the sleep that is controlled by this value will be removed entirely in the subsequent PR, where a proper wait mechanism, based on release status, is implemented. I'd just leave it as it is for now.